ghcup-vanilla-0.0.7.yaml

```diff
@@ -5804,6 +5803,101 @@ ghcupDownloads:
+            '(>= 20 && < 21)':
+              dlUri: https://downloads.haskell.org/~hls/haskell-language-server-2.6.0.0/haskell-language-server-2.6.0.0-x86_64-linux-mint20.2.tar.xz
+              dlSubdir: haskell-language-server-2.6.0.0
+              dlHash: 8cfd87f928a40fcdebc2ed40407c9ffdc974fb3135de066b3250f542b46af727
+            '>= 21': *hls-2600-64-ubuntu22
+          Linux_Fedora:
+            '< 33': &hls-2600-64-fedora27
+              dlUri: https://downloads.haskell.org/~hls/haskell-language-server-2.6.0.0/haskell-language-server-2.6.0.0-x86_64-linux-fedora27.tar.xz
+              dlSubdir: haskell-language-server-2.6.0.0
+              dlHash: faf23c3276584577ebeef4e2f4d23b6f27bfb1921f9c3fb8d3a2f72fa5ce6c24
+            '>= 33': &hls-2600-64-fedora33
+              dlUri: https://downloads.haskell.org/~hls/haskell-language-server-2.6.0.0/haskell-language-server-2.6.0.0-x86_64-linux-fedora33.tar.xz
+              dlSubdir: haskell-language-server-2.6.0.0
+              dlHash: 10cbdeececd8c31f9367e8116d6e2d89469900340b9c999ffe84fb03bf31fab7
+            unknown_versioning: *hls-2600-64-fedora27
+          Linux_CentOS:
+            '( >= 7 && < 8 )': &hls-2600-64-centos
+              dlUri: https://downloads.haskell.org/~hls/haskell-language-server-2.6.0.0/haskell-language-server-2.6.0.0-x86_64-linux-centos7.tar.xz
+              dlSubdir: haskell-language-server-2.6.0.0
+              dlHash: 2c90d2c4525bf36936b0ef505760bf071cc4350fc9205146d54fbac2d2295085
+            unknown_versioning: *hls-2600-64-centos
+          Linux_RedHat:
+            unknown_versioning: *hls-2600-64-centos
+          Linux_UnknownLinux:
+            unknown_versioning:
+              dlUri: https://downloads.haskell.org/~hls/haskell-language-server-2.6.0.0/haskell-language-server-2.6.0.0-x86_64-linux-unknown.tar.xz
+              dlSubdir: haskell-language-server-2.6.0.0
+              dlHash: 9341212f1b9215394dca18fa47e19820728ac62002c28438c1a628089446c22c
+          Darwin:
+            unknown_versioning:
+              dlUri: https://downloads.haskell.org/~hls/haskell-language-server-2.6.0.0/haskell-language-server-2.6.0.0-x86_64-apple-darwin.tar.xz
+              dlSubdir: haskell-language-server-2.6.0.0
+              dlHash: da96a6bfd30b5a6daee21d712304bd3732a4f1a94956a478720bc6652fbde9ea
+          Windows:
+            unknown_versioning:
+              dlUri: https://downloads.haskell.org/~hls/haskell-language-server-2.6.0.0/haskell-language-server-2.6.0.0-x86_64-mingw64.zip
+              dlHash: 6ba6eb86d0936ab3e5639b928c43d69990087ea68a801c9b816a98d3d2312d43
+          FreeBSD:
+            unknown_versioning:
+              dlUri: https://downloads.haskell.org/~hls/haskell-language-server-2.6.0.0/haskell-language-server-2.6.0.0-x86_64-freebsd.tar.xz
+              dlSubdir: haskell-language-server-2.6.0.0
+              dlHash: 8d172261b2b2b2fb09572589b192245a5e306e34f3f4e2b92af4fdb565564a99
```

Comment:
This is erroneous.

The vanilla channel does not support the FreeBSD GHC versions that this HLS bindist supplies. You're building the HLS bindists against the main channel, but then add it to the vanilla-channel, which doesn't add up.

You have to build against the vanilla channel in HLS release CI, so that the configuration matches properly.

@michaelpj

Reply:
> You have to build against the vanilla channel in HLS release CI, so that the configuration matches properly.

Then these bindists will only be usable by people using the vanilla channel. Why even build HLS bindists officially in that case? It seems like you are suggesting that ghcup will take the responsibility of building all HLS bindists.

Reply:
> Then these bindists will only be usable by people using the vanilla channel.

Well, the channel has to be self-consistent. It is an alternative to the default channel. It's not meant to be added on top of the default channel.

> It seems like you are suggesting that ghcup will take the responsibility of building all HLS bindists.

Yes, that's what I'm planning to do: https://github.com/haskellfoundation/tech-proposals/pull/61

Reply:
We will build HLS bindists against the vanilla channel next time, but it is basically impossible to distribute those bindists as switching channels is not safe.

Reply:
This is how all linux distributions with different sources work.

There is no plan to make more complicated abstractions to support users randomly switching between channels.

HLS already does ABI checks on startup.

GHCup maintainers discourage the use of the vanilla channel for reasons explained in the tech proposal.